### PR TITLE
Fix #207 SfAddToCart

### LINKED ISSUE #207: [ Desktop / Mobile ] Implement SfAddToCart molecule
image
image

Proposed api
```html
<SfAddToCart v-model="qty" @click="addToCart(product, qty)" />
```

If you have any suggestions to make this API better don't hesitate to propose it

### PR DESCRIPTION
**Pull request by mayashavin**
# Related issue
[207](https://github.com/DivanteLtd/storefront-ui/issues/207)

# Scope of work
1. Add `SfAddToCart` with support for `slot`
2. Add unit test

# Screenshots of visual changes

Screen Shot 2019-08-23 at 17 28 25

With `slot` for quantity select input design:

Screen Shot 2019-08-23 at 17 28 46

With `slot` for Add to Cart button
Screen Shot 2019-08-23 at 17 28 38

# Checklist

- [x] I followed [composition rules](https://github.com/DivanteLtd/storefront-ui/blob/master/docs/component-rules.md) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/shared/styles/components/SfAddToCart.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    margin-left: 0.5rem;
+
+    input {
+      padding-left: 0.5rem;
+      padding-right: 0.5rem;
+      width: auto;
+      height: 100%;
+      max-width: 90px;
+      border: 1px solid $c-light-primary;
+      transition: all 0.3s ease-in-out;
```
Comment (przspa) [inline packages/shared/styles/components/SfAddToCart.scss]:
```suggestion
      transition: border-color 0.3s ease-in-out;
```
is more efficient when you used only property with you change :)

File: packages/shared/styles/components/SfAddToCart.scss [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  &__select-quantity {
+    margin-left: 0.5rem;
+
+    input {
+      padding-left: 0.5rem;
+      padding-right: 0.5rem;
+      width: auto;
+      height: 100%;
+      max-width: 90px;
```
Comment (przspa) [inline packages/shared/styles/components/SfAddToCart.scss]: meybe we need scss variable for it?

File: packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        Add to cart
+    </SfButton>
+  </slot>
+  <slot name="quantity-select-input" v-bind="{ qty }">
+    <SfInput 
+      :value="qty" 
+      v-on:change="$emit('input', qty)"
+      class="sf-add-to-cart__select-quantity"
+      type="number"
+      min=1
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.html]:
```suggestion
      min="1"
```

File: packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    />
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.html]:
```suggestion
    :max="quantity"
    />
```

File: packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  <slot name="add-to-cart-btn">
+    <!--@slot Custom content that will replace default Add to cart button design.-->
+    <SfButton v-on="$listeners" class="sf-add-to-cart__button" :disabled="!canAddToCart">
+        Add to cart
+    </SfButton>
+  </slot>
+  <slot name="quantity-select-input" v-bind="{ qty }">
+    <SfInput 
+      :value="qty" 
+      v-on:change="$emit('input', qty)"
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.html]:
```suggestion
    @input="(value)=>{$emit('input', value)}"
```
becouse SfInput use v-model :) and we shoud use :value and @input

File: packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.html [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<div class="sf-add-to-cart" data-test="sf-add-to-card">
+  <slot name="add-to-cart-btn">
+    <!--@slot Custom content that will replace default Add to cart button design.-->
+    <SfButton v-on="$listeners" class="sf-add-to-cart__button" :disabled="!canAddToCart">
+        Add to cart
+    </SfButton>
+  </slot>
+  <slot name="quantity-select-input" v-bind="{ qty }">
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.html]:
```suggestion
  <slot name="quantity-select-input" v-bind="{ qty, quantity }">
```

File: packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+     * can be added to cart
+     */
+    canAddToCart: {
+      type: Boolean,
+      default: false
+    },
+    /**
+     * Stock quantity of product
+     */
+    quantity: {
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.js]:
```suggestion
    stock: {
```
maybe we should name it stock, stockQuantity or something like that

Comment (mayashavin) [reply packages/vue/src/components/molecules/SfAddToCart/SfAddToCart.js]: stock is good :)

File: packages/shared/styles/components/SfAddToCart.scss (hunk 2 above)
Comment (mayashavin) [reply packages/shared/styles/components/SfAddToCart.scss]: for `max-width`? Sure :)

Comment (przspa) [reply packages/shared/styles/components/SfAddToCart.scss]: for my yes but if you think is not necessary then feel free :)

**Pull request merged**